src/main/java/address/Browser/BrowserManager.java

```diff
@@ -1,53 +1,177 @@
+import address.model.Person;
 import com.teamdev.jxbrowser.chromium.Browser;
 import com.teamdev.jxbrowser.chromium.EditorCommand;
 import com.teamdev.jxbrowser.chromium.dom.By;
 import com.teamdev.jxbrowser.chromium.dom.DOMElement;
 import com.teamdev.jxbrowser.chromium.dom.events.DOMEventType;
 import com.teamdev.jxbrowser.chromium.events.FinishLoadingEvent;
 import com.teamdev.jxbrowser.chromium.events.LoadAdapter;
+import javafx.scene.control.TabPane;
+
+import java.util.List;
 
 /**
  * Manages the browser.
  */
 public class BrowserManager {
 
-    private Browser browser;
+    private static final int PERSON_DOES_NOT_EXISTS_IN_BROWSER = -1;
+    private static final int NUMBER_OF_PRELOADED_PAGE = 3;
+    public static final String INITIAL_PAGE_HTML_CODE =
+            "<html><body><h3>To view contact's web page, click on the contact on the left.</h3></body></html>";
+
+    private TabBrowser browser;
 
-    public BrowserManager() {
-        this.browser = new Browser();
+    private ModelManager modelManager;
+
+    public BrowserManager(ModelManager modelManager) {
+        this.modelManager = modelManager;
+        browser = new TabBrowser(NUMBER_OF_PRELOADED_PAGE);
         registerListeners();
     }
 
     private void registerListeners() {
-        this.browser.addLoadListener(new LoadAdapter() {
-            @Override
-            public void onFinishLoadingFrame(FinishLoadingEvent finishLoadingEvent) {
-                automateClickingAndScrolling();
-            }
-        });
+        for (int i = 0; i < browser.getNumberOfPages(); i++) {
+            browser.getPage(i).addLoadListener(new LoadAdapter() {
+                @Override
+                public void onFinishLoadingFrame(FinishLoadingEvent finishLoadingEvent) {
+                    if (finishLoadingEvent.isMainFrame()) {
+                        automateClickingAndScrolling(finishLoadingEvent.getBrowser());
+                        System.out.println("finish loading frame");
+                    }
+                }
+            });
+        }
     }
 
     /**
-     * Frees resources allocated to the browser.
+     * Initialise browser to initial state.
      */
-    public void freeBrowserResources(){
-        browser.dispose();
+    public void initBrowser() {
+        for (int i = 0; i < browser.getNumberOfPages(); i++){
+            browser.getPage(i).loadHTML(INITIAL_PAGE_HTML_CODE);
+        }
     }
 
-    public Browser getBrowser(){
-        return browser;
+    /**
+     * Loads the person's GitHub profile page to the browser.
+     * PreCondition: ModelManager getFilteredPersons method returns more than one person.
+     * @param person
+     */
+    public void loadGitHubProfilePage(Person person) {
+
+        List<Person> filteredPersons = modelManager.getFilteredPersons();
+        int indexOfContactsInFilteredListView = filteredPersons.size();
+
+        int indexOfSelectedTabBrowserPage = findIndexOfTabBrowserThatContainsPersonGitHubWebPage(person);
```

Comment:
This should be `Optional` instead of having to compare it with a magic number if it doesn't exist.